cpp/src/parquet/arrow/test_util.h

```diff
@@ -142,7 +142,11 @@ template <int32_t byte_width>
 static void random_decimals(int64_t n, uint32_t seed, int32_t precision, uint8_t* out) {
   auto gen = ::arrow::random::RandomArrayGenerator(seed);
   std::shared_ptr<Array> decimals;
-  if constexpr (byte_width == 16) {
+  if constexpr (byte_width == 4) {
+    decimals = gen.Decimal32(::arrow::decimal32(precision, 0), n);
+  } else if constexpr (byte_width == 8) {
+    decimals = gen.Decimal64(::arrow::decimal32(precision, 0), n);
```

Comment:
why ::arrow::decimal32?

Reply:
presumably to disambiguate `parquet::arrow` and `arrow` namespaces

Reply:
I mean when bitwidth==8, why it's a decimal32 here 🤔?

Reply:
ah...good catch